### PR TITLE
license info output as xhtml and docx

### PR DESCRIPTION
**Pull request by bs-jokri**

### CODE REVIEW

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java [hunk 1]
```diff
@@ -150,16 +157,55 @@ public LicenseInfoParsingResult getLicenseInfoForRelease(Release release) throws
...
+                    generator.getOutputFormatInfo()
+            );
+        }
+        return outputPossibilities;
+    }
+
+    @Override
+    public OutputFormatInfo getOutputFormatInfoForGeneratorClass(String generatorClassName) throws TException{
+        for (OutputGenerator generator : outputGenerators) {
+            if (generatorClassName.equals(generator.getClass().getName())) {
```
Comment (alex-evo) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java]: This loop and comparison of generator's class name occurs three times in this file. I think it's time to introduce a method `findGeneratorByClassName` or something like that.

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/common/PortalConstants.java [hunk 2]
```diff
@@ -157,6 +161,8 @@
     public static final String WHERE_ARRAY = "where[]";
     public static final String HOW = "how";
 
+
+
```
Comment (alex-evo) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/common/PortalConstants.java]: this must be a glitch

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 3]
```diff
@@ -129,15 +128,26 @@ public void serveResource(ResourceRequest request, ResourceResponse response) th
...
-            PortletResponseUtil.sendFile(request, response, fileName, licenseInfoClient.getLicenseInfoFileForProject(projectId, user).getBytes(), "text/plain");
+            String projectName = project != null ? project.getName() : "Unknown-Project";
+            String timestamp = SW360Utils.getCreatedOn();
+            OutputFormatInfo outputFormatInfo = licenseInfoClient.getOutputFormatInfoForGeneratorClass(generatorClassName);
+            String filename = "LicenseInfo-" + projectName + "-" + timestamp + "." + outputFormatInfo.getFileExtension();
+            if(outputFormatInfo.isOutputBinary){
+                ByteBuffer licenseInfoByteBuffer = licenseInfoClient.getLicenseInfoFileForProjectAsBinary(projectId, user, generatorClassName);
+                byte[] licenseInfoByteArray = new byte[licenseInfoByteBuffer.remaining()];
+                licenseInfoByteBuffer.get(licenseInfoByteArray);
+                PortletResponseUtil.sendFile(request, response, filename, licenseInfoByteArray, "text/plain");
```
Comment (alex-evo) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/projects/ProjectPortlet.java]: I have doubts that MIME type text/plain is correct

File: frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/projects/ProjectPortlet.java [hunk 4]
```diff
@@ -288,7 +298,7 @@ private void serveNewTableRowLinkedRelease(ResourceRequest request, ResourceResp
             ComponentService.Iface client = thriftClients.makeComponentClient();
             for (Release release : client.getReleasesById(new HashSet<>(Arrays.asList(linkedIds)), user)) {
                 final Vendor vendor = release.getVendor();
-                final String fullname = vendor!=null?vendor.getFullname():"";
+                final String fullname = vendor != null ? vendor.getFullname() : "";
```
Comment (alex-evo) [inline frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/projects/ProjectPortlet.java]: would you please stop it? pretty please?

File: libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift [hunk 5]
```diff
@@ -56,6 +69,20 @@ service LicenseInfoService {
...
+
+    binary getLicenseInfoFileForProjectAsBinary(1: string projectId, 2: User user, 3: string outputType);
+
+    /**
+      * returns all available output types
+      */
+    list<OutputFormatInfo> getPossibleOutputFormats();
+
+    /**
+     * returns file extension that is applicable when licenseInfo is generated by the generator class
```
Comment (alex-evo) [inline libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift]: doc comment isn't quite correct. the method returns more than just the file extension.

File: libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift [hunk 6]
```diff
@@ -23,12 +23,25 @@ enum LicenseInfoRequestStatus{
     FAILURE = 2,
 }
 
+struct OutputFormatInfo{
```
Comment (alex-evo) [inline libraries/lib-datahandler/src/main/thrift/licenseinfo.thrift]: maybe makes sense to extend with `string mimetype`

**Comment by heydenreich** [comment]
Rebased, result can be found on rebase/licenseInfoOutput. bsinno-branch has to be updated to update this PR.

**Comment by mcjaeger** [comment]
found while testing https://github.com/sw360/sw360portal/issues/238
seems to be the same as https://github.com/sw360/sw360portal/issues/217

**Pull request merged**